onnxmltools/convert/sklearn/operator_converters/NaiveBayes.py

```diff
@@ -201,23 +226,27 @@ def convert_sklearn_naive_bayes(scope, operator, container):
                            neg_prob_name, name=scope.get_unique_operator_name('Log'))
         container.add_node('ReduceSum', neg_prob_name,
                            sum_neg_prob_name, name=scope.get_unique_operator_name('ReduceSum'), axes=[0])
-        container.add_node('MatMul', [input_name, neg_prob_name],
-                           inp_neg_prob_prod_name, name=scope.get_unique_operator_name('MatMul'))
-        # Cast is required here as Sub op doesn't work with Float64
-        container.add_node('Cast', inp_neg_prob_prod_name, 
-                            cast_result_name, to=onnx_proto.TensorProto.FLOAT, op_version=7)
-        apply_sub(scope, [sum_neg_prob_name, cast_result_name], difference_matrix_name, container, broadcast=1)
-        container.add_node('Sum', [difference_matrix_name, class_log_prior_name],
+        apply_sub(scope, [feature_log_prob_name, neg_prob_name], difference_matrix_name, container)
+        container.add_node('MatMul', [input_name, difference_matrix_name],
+                           dot_prod_name, name=scope.get_unique_operator_name('MatMul'))
+        container.add_node('Sum', [sum_neg_prob_name, dot_prod_name],
+                           partial_sum_result_name, name=scope.get_unique_operator_name('Sum'))
+        container.add_node('Sum', [partial_sum_result_name, class_log_prior_name],
                            sum_result_name, name=scope.get_unique_operator_name('Sum'))
 
     container.add_node('ArgMax', sum_result_name,
                        argmax_output_name, name=scope.get_unique_operator_name('ArgMax'), axis=1)
 
     # Following four statements are for predicting probabilities
```

Comment:
Which four statements are you referring here?

Reply:
Those were the four add_node() / apply_* statements to calculate prob_tensor_name (245, 248, 249 and 250). I'll remove the confusing comment and replace it with "Calculation of class probability".